File: prueba-frontend/src/app/modules/intranet/services/factura.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { FacturaBuscar } from '../dto/request/factura-buscar';
import { Factura } from '../dto/response/factura';
import { OutResponse } from '../dto/response/out.response';

@Injectable()
export class FacturaService {

  constructor(private http: HttpClient) { }

  public registrarFactura(req: Factura): Observable<OutResponse<Factura>> {
    return this.http.post<OutResponse<Factura>>(`${environment.pruebaBackendUrl}/factura/registrarFactura`, req);
  }

  public modificarFactura(req: Factura): Observable<OutResponse<any>> {
    return this.http.post<OutResponse<any>>(`${environment.pruebaBackendUrl}/factura/modificarFactura`, req);
  }

  public listarFactura(req: Factura): Observable<OutResponse<Factura[]>> {
    return this.http.post<OutResponse<Factura[]>>(`${environment.pruebaBackendUrl}/factura/listarFactura`, req);
  }

  public eliminarFacturaLogica(req: Factura): Observable<OutResponse<any>> {
    return this.http.post<OutResponse<any>>(`${environment.pruebaBackendUrl}/factura/eliminarFacturaLogica`, req);
  }

  public listarFacturaPaginado(req: FacturaBuscar): Observable<OutResponse<Factura[]>> {
    return this.http.post<OutResponse<Factura[]>>(`${environment.pruebaBackendUrl}/factura/listarFacturaPaginado`, req);
  }

  public buscarFactura(req: Factura): Observable<OutResponse<Factura>> {
    return this.http.post<OutResponse<Factura>>(`${environment.pruebaBackendUrl}/factura/buscarFactura`, req);
  }
}

File: prueba-backend/src/main/java/com/empresa/proyecto/service/FacturaService.java
package com.empresa.proyecto.service;

import java.util.List;

import com.empresa.proyecto.dto.request.FacturaBuscar;
import com.empresa.proyecto.dto.response.Factura;
import com.empresa.proyecto.dto.response.OutResponse;

public interface FacturaService {

	public OutResponse<Factura> registrarFactura(Factura c) throws Exception;

	public OutResponse<?> modificarFactura(Factura c) throws Exception;

	public OutResponse<List<Factura>> listarFactura(Factura r);

	public OutResponse<?> eliminarFacturaLogica(Factura r);

	public OutResponse<?> eliminarFactura(Factura c);

	public OutResponse<List<Factura>> listarFacturaPaginado(FacturaBuscar r);
	
	public OutResponse<Factura> buscarFactura(Factura c);
}
